service/history/ndc/history_replicator.go

```diff
@@ -104,6 +105,7 @@ type (
 		workflowCache     wcache.Cache
 		eventsReapplier   EventsReapplier
 		transactionMgr    TransactionManager
+		rateLimiter       quotas.RateLimiter
```

Comment:
We may want to have a dedicated type for this rate limiter. Similar to https://github.com/temporalio/temporal/blob/1bf96bd729d733195449716d27631ae4c95232b4/service/history/replication/quotas.go#L12